### ISSUE TITLE
PostgreSQL bad execution plan

### ISSUE THREAD

**Issue by rjtokenring** [prestodb/presto#7390]
Hi!
While using postgresql driver I can see that on some queries a wrong binding type is performed. 
Query, as logged in pg_logs, is:

< 2017-02-16 16:57:23.199 CET > LOG:  duration: 0.912 ms  parse S_1: SELECT "objid" FROM "public"."obj" WHERE (("objid" > $1 AND "objid" < $2))
< 2017-02-16 16:57:23.200 CET > LOG:  duration: 0.663 ms  bind S_1/C_2: SELECT "objid" FROM "public"."ob" WHERE (("objid" > $1 AND "objid" < $2))
< 2017-02-16 16:57:23.200 CET > DETAIL:  parameters: $1 = '2334', $2 = '2456'
< 2017-02-16 16:57:29.301 CET > LOG:  duration: 6100.479 ms  execute S_1/C_2: SELECT "objid" FROM "public"."obj" WHERE (("objid" > $1 AND "objid" < $2))
< 2017-02-16 16:57:29.301 CET > DETAIL:  parameters: $1 = '2334', $2 = '2456'

Same query, with proper binding, it takes less than 200ms to be executed. In this case it is executed in more than 6s. This is compatible with a pg query executed with wrong parameter type (eg forcing param as varchar, while here column objid is numeric). Table is indexed by objid numeric and index is used properly if query is executed from a sql client.

This using presto 0.163

**Comment by cawallin** [comment]
@rjtokenring -- it's a little hard to follow the above example, what is the query and the schema of the table?
